master/internal/rm/kubernetesrm/pod_test.go

```diff
@@ -179,14 +176,15 @@ func TestResourceCreationFailed(t *testing.T) {
 	require.NoError(t, err)
 	time.Sleep(time.Second)
 
-	// We expect two messages in the queue because the pod actor sends itself a stop message.
-	assert.Equal(t, sub.Len(), 2)
+	// We expect three messages in the queue because the pod actor sends itself a stop message.
+	assert.Equal(t, sub.Len(), 3)
 	message := sub.Get()
-	containerMsg, ok := message.(*sproto.ContainerLog)
-	if !ok {
-		t.Errorf("expected sproto.ContainerLog but received %s", reflect.TypeOf(message))
-	}
-	assert.ErrorContains(t, errors.New(*containerMsg.AuxMessage), correctMsg)
+	require.IsType(t, &sproto.ContainerLog{}, message)
+	require.Contains(t, *message.(*sproto.ContainerLog).AuxMessage, "Pod Spec Configured")
+
+	message = sub.Get()
+	require.IsType(t, &sproto.ContainerLog{}, message)
+	require.Contains(t, *message.(*sproto.ContainerLog).AuxMessage, "already exists")
```

Comment:
part of k8s improvements should definitely be improving these tests. this is really fragile.

Reply:
like I don't even know what this test is really testing anyway at this point, the error comes from our mock...

I don't know

They have just been like touched just enough so they pass on every change so I think they are basically meaningless at this point